### ISSUE TITLE
Bug: Failure in DataLoadMesh Access

### ISSUE THREAD

**Issue by yoqim** [mvp-ai-lab/mvp-dataset#18]
## Summary

When using `mvp-dataset` with a PyTorch `DeviceMesh` whose mesh dimensions are named, distributed data loading fails while computing `DataLoadMesh.dp_rank`.

The failure happens because `DataLoadMesh` calls:

```python
self.device_mesh.size(dim)
```
where dim is a string such as "replicate". In PyTorch 2.8, this call reaches the underlying mesh tensor, whose tensor dimension names are None, so "replicate" cannot be resolved.

## Traceback
Traceback points to: mvp_dataset/core/mesh.py
```python
sizes = [self.device_mesh.size(dim) for dim in self.dp_dims]
```

## Suggested Fix

Map string mesh dimension names to integer indices using device_mesh.mesh_dim_names before calling size().
For local rank, avoid relying on process groups inside DataLoader workers. device_mesh.get_local_rank(dim) may fail in workers because it resolves process groups. device_mesh.get_coordinate() is safer when available.
